workspaces/ui-v2/src/components/sharing/ShareModal.tsx

```diff
@@ -0,0 +1,223 @@
+  const params = {
+    label: 'API Docs',
+    message: `${endpointCount} Endpoints`,
+    style: 'flat',
+    color: 'rgb(43,123,209)',
+    // Took https://github.com/opticdev/optic/blob/develop/website/static/img/logo-bare.svg
+    // Ran it through https://jakearchibald.github.io/svgomg/
+    // base64 encoded it (on macos, you can do: `pbpaste | base64 | pbcopy` to quickly base64 encode whatever is in your paste buffer)
+    // prepend with `data:image/svg+xml;base64,`
+    logo:
+      'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTM0IiBoZWlnaHQ9IjEzNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+PGRlZnM+PGxpbmVhckdyYWRpZW50IHgxPSI1MCUiIHkxPSItMTQuMyUiIHgyPSI1MCUiIHkyPSIxMDAlIiBpZD0iYyI+PHN0b3Agc3RvcC1jb2xvcj0iI0ZGRiIgb2Zmc2V0PSIwJSIvPjxzdG9wIHN0b3AtY29sb3I9IiNGNEY0RjQiIG9mZnNldD0iMTAwJSIvPjwvbGluZWFyR3JhZGllbnQ+PGZpbHRlciB4PSItNS42JSIgeT0iLTQlIiB3aWR0aD0iMTExLjElIiBoZWlnaHQ9IjExMS4xJSIgZmlsdGVyVW5pdHM9Im9iamVjdEJvdW5kaW5nQm94IiBpZD0iYSI+PGZlT2Zmc2V0IGR5PSIyIiBpbj0iU291cmNlQWxwaGEiIHJlc3VsdD0ic2hhZG93T2Zmc2V0T3V0ZXIxIi8+PGZlR2F1c3NpYW5CbHVyIHN0ZERldmlhdGlvbj0iMiIgaW49InNoYWRvd09mZnNldE91dGVyMSIgcmVzdWx0PSJzaGFkb3dCbHVyT3V0ZXIxIi8+PGZlQ29sb3JNYXRyaXggdmFsdWVzPSIwIDAgMCAwIDAgMCAwIDAgMCAwIDAgMCAwIDAgMCAwIDAgMCAwLjUgMCIgaW49InNoYWRvd0JsdXJPdXRlcjEiLz48L2ZpbHRlcj48Y2lyY2xlIGlkPSJiIiBjeD0iNjMiIGN5PSI2MyIgcj0iNjMiLz48L2RlZnM+PGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj48ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHRyYW5zZm9ybT0idHJhbnNsYXRlKDQgMikiPjx1c2UgZmlsbD0iIzAwMCIgZmlsdGVyPSJ1cmwoI2EpIiB4bGluazpocmVmPSIjYiIvPjx1c2UgZmlsbD0idXJsKCNjKSIgeGxpbms6aHJlZj0iI2IiLz48L2c+PHBhdGggZD0iTTkwIDM1YTM4IDM4IDAgMDAtNDYgMG0yOCA2OGEzOCAzOCAwIDAwMzMtMzhjMC04LTMtMTYtNy0yMm0tMzYgNjBhMzggMzggMCAwMS0zMy0zOGMwLTggMy0xNiA3LTIyIiBzdHJva2U9IiMzNDNCRTUiIHN0cm9rZS13aWR0aD0iNC43IiBzdHJva2UtbGluZWNhcD0icm91bmQiLz48cGF0aCBkPSJNNTEgODVhMjYgMjYgMCAwMDMyIDBNNjMgMzljLTUgMS0xMCAzLTE0IDdhMjYgMjYgMCAwMC0zIDM0bTI1LTQxYzUgMSAxMCAzIDE0IDdhMjYgMjYgMCAwMTMgMzQiIHN0cm9rZT0iIzM0M0JFNSIgc3Ryb2tlLXdpZHRoPSI0LjciIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIvPjwvZz48L3N2Zz4=',
+  };
+
+  const queryParams = Object.entries(params)
+    .map(([k, v]) => `${k}=${encodeURIComponent(v)}`)
+    .join('&');
+
+  return `https://img.shields.io/static/v1?${queryParams}`;
+}
+
+function generateBadgeMarkdown({
+  endpointCount,
+  shareUrl,
+}: {
+  endpointCount: number;
+  shareUrl: string;
+}) {
+  return `[![API Docs: ${endpointCount} Endpoints](${generateBadgeUrl({
+    endpointCount,
+  })})](${shareUrl})`;
+}
+
+export const ShareModal: React.FC<{
+  share: (intent: ShareTarget) => Promise<string>;
+}> = (props) => {
+  const styles = useStyles(props);
+
+  const { isAuthenticated, isLoading, loginWithPopup } = useAuth0();
+
+  const endpointCount = useAppSelector(
+    (state) => state.endpoints.results?.data!.length || 0
+  );
+
+  const [shareType, setShareType] = useState<ShareTarget | null>(null);
+
+  let inputRef = useRef<HTMLInputElement>();
```

Comment:
We probably don't need to use an input tag here - we already have the shareUrl as state, so we can use that instead of overriding the user's focus + select state (probably not a big deal in this specific example) - we could build a copyToClipboard util function here in place of this

Reply:
We talked in a DM about this, I think tentatively came to the conclusion that changing focus to select the text that you clicked on a button to copy seems okay/expected behavior. Do we still want to do this "invisibly"? Happy to do either, I don't really have a strong opinion -- my weak opinion is that this works fine, and doesn't seem to be that jarring

Reply:
Yeah I think that's fine, it's not a huge deal. These interactions are minor - at some point we'll have some UI / UX patterns + consistency where we'll want to have the same behavior across the app